src/lib/prover/prover.ml

```diff
@@ -269,5 +267,28 @@ let initialized {connection; _} =
 
 let extend_blockchain {connection; _} chain next_state block prover_state
     pending_coinbase =
-  Worker.Connection.run connection ~f:Worker.functions.extend_blockchain
-    ~arg:(chain, next_state, block, prover_state, pending_coinbase)
+  let input =
+    { Extend_blockchain_input.chain
+    ; next_state
+    ; block
+    ; prover_state
+    ; pending_coinbase }
+  in
+  match%map
+    Worker.Connection.run connection ~f:Worker.functions.extend_blockchain
+      ~arg:input
+    >>| Or_error.join
+  with
+  | Ok x ->
+      Ok x
+  | Error e ->
+      Logger.error (Logger.create ()) ~module_:__MODULE__ ~location:__LOC__
+        ~metadata:
+          [ ( "input-sexp"
+            , `String
+                (Sexp.to_string (Extend_blockchain_input.sexp_of_t input)) )
+          ; ( "input-bin-io"
```

Comment:
Do we need both of these? It seems a bit redundant to output both when the represent the same data.